dts/bindings/adc/adc-controller.yaml

```diff
@@ -12,3 +13,136 @@ properties:
     "#io-channel-cells":
       type: int
       required: true
+
+    "#address-cells":
+      required: false
+      const: 1
+
+    "#size-cells":
+      required: false
+      const: 0
+
+child-binding:
+    description: |
+      Channel configuration.
+
+      All nodes using this binding must be named "channel", otherwise their
+      data will not be accessible for the ADC API macros.
+
+      This is based on Linux, documentation:
+        https://www.kernel.org/doc/Documentation/devicetree/bindings/iio/adc/adc.yaml
+
+    properties:
+      reg:
+        type: array
+        required: true
+        description: Channel identifier.
+
+      zephyr,gain:
+        type: string
+        required: true
+        description: |
+          Gain selection:
+          - ADC_GAIN_1_6: x 1/6
+          - ADC_GAIN_1_5: x 1/5
+          - ADC_GAIN_1_4: x 1/4
+          - ADC_GAIN_1_3: x 1/3
+          - ADC_GAIN_2_5: x 2/5
+          - ADC_GAIN_1_2: x 1/2
+          - ADC_GAIN_2_3: x 2/3
+          - ADC_GAIN_4_5: x 4/5
+          - ADC_GAIN_1:   x 1
+          - ADC_GAIN_2:   x 2
+          - ADC_GAIN_3:   x 3
+          - ADC_GAIN_4:   x 4
+          - ADC_GAIN_6:   x 6
+          - ADC_GAIN_8:   x 8
+          - ADC_GAIN_12:  x 12
+          - ADC_GAIN_16:  x 16
+          - ADC_GAIN_24:  x 24
+          - ADC_GAIN_32:  x 32
+          - ADC_GAIN_64:  x 64
+          - ADC_GAIN_128: x 128
+        enum:
+          - "ADC_GAIN_1_6"
```

Comment:
Why not have these be more readable since they are strings?

`"1/6"`
`"1/5"`
...